dali/pipeline/operator/sequence_operator.h

```diff
@@ -0,0 +1,451 @@
+                                              const std::string &arg_name,
+                                              const ExpandDesc &expand_desc) {
+    DALI_ENFORCE(
+        expand_desc.HasFrames(),
+        make_string(
+            "Tensor input for argument ", arg_name, " is specified per frame (got ",
+            arg_tensor.GetLayout(),
+            " layout), but samples in the input batch do not contain frames (expected input "
+            "layout that starts with 'F', ",
+            ShouldExpandChannels() ? " or 'CF'" : "", ")."));
+    const auto &shape = arg_tensor.shape();
+    int arg_sample_dim = shape.sample_dim();
+    TensorVector<CPUBackend> flat_tensor(expand_desc.NumElements());
+    int num_samples = expand_desc.NumSamples();
+    auto type_info = arg_tensor.type_info();
+    auto type = type_info.id();
+    auto is_pinned = arg_tensor.is_pinned();
+    auto order = arg_tensor.order();
+    int elem_idx = 0;
+    for (int sample_idx = 0; sample_idx < num_samples; sample_idx++) {
+      const auto &sample_shape = shape[sample_idx];
+      int num_input_frames = expand_desc.NumFrames(sample_idx);
+      int num_arg_frames = sample_shape[0];
+      DALI_ENFORCE(
+          num_arg_frames == 1 || num_input_frames == num_arg_frames,
+          make_string("The tensor argument ", arg_name, " for sample ", sample_idx,
+                      " should either be a single argument to be reused accross all frames in the "
+                      "sample or should be specified per each frame. Got ",
+                      num_arg_frames, " arguments for the sample but there are ", num_input_frames,
+                      " frames in the sample."));
+      auto elem_shape = sample_shape.last(arg_sample_dim - 1);
+      auto elem_volume = volume(elem_shape);
+      uint8_t *base_ptr =
+          const_cast<uint8_t *>(static_cast<const uint8_t *>(arg_tensor.raw_tensor(sample_idx)));
+      auto num_bytes = type_info.size() * elem_volume;
+      if (num_arg_frames == 1) {  // broadcast the sample
+        for (int j = 0; j < expand_desc.NumElements(sample_idx); j++) {
+          flat_tensor[elem_idx++].ShareData(base_ptr, num_bytes, is_pinned, elem_shape, type,
+                                            order);
+        }
+      } else if (!expand_desc.HasChannels()) {  // expand frames dimension
+        assert(num_arg_frames == expand_desc.NumElements(sample_idx));
+        for (int i = 0; i < num_arg_frames; i++, base_ptr += num_bytes) {
+          flat_tensor[elem_idx++].ShareData(base_ptr, num_bytes, is_pinned, elem_shape, type,
+                                            order);
+        }
+      } else {
+        int num_input_channels = expand_desc.NumChannels(sample_idx);
+        assert(num_arg_frames * num_input_channels == expand_desc.NumElements(sample_idx));
+        int inner_stride, outer_stride;
+        if (expand_desc.IsChannelFirst()) {
```

Comment:
This looks like expansion over channels is not optional. I think it should be `expand_desc_.ExpandChannels()`, not `IsChannelFirst`.